test/models/ab_test_test.rb

```diff
@@ -0,0 +1,89 @@
+require 'test_helper'
+
+class AbTestTest < ActiveSupport::TestCase
+
+  before(:all) do
+    @user = FactoryBot.create(:user, test_array: @@users_to_clean)
+    @feature_flag = FeatureFlag.create(name: 'ab_test_flag')
+    # random number of group names from 2-5
+    limit = (2..5).to_a.sample
+    group_names = 1.upto(limit).map { |i| "name-#{i}" }
+    @ab_test = AbTest.create(feature_flag: @feature_flag, group_names:)
+  end
+
+  teardown do
+    @ab_test.update(enabled: false)
+    @ab_test.reload
+    @user.feature_flag_options.destroy_all
+  end
+
+  after(:all) do
+    @feature_flag.destroy
+  end
+
+  # we can't prove true random assignment, only that over a reasonable amount of iterations we would see a roughly
+  # equal distribution (±2%); for reference, SCP had ~2.5K unique Study overview users/week in 2022
```

Comment:
I can't help myself from linking to this statistic: https://mixpanel.com/s/1hzUx.